### ISSUE TITLE
Something broke SafeHaskell again.

### ISSUE THREAD

**Issue by ekmett** [ekmett/lens#563]
`Control.Lens` doesn't infer as `Safe`.

**Comment by ekmett** [comment]
Since 4.11

**Comment by ekmett** [comment]
It seems that in certain build configurations it fails to build Safe on 4.10 as well.

**Comment by glguy** [comment]
Some of the build failures with Safe have to do with the doctest portion of the build. (I haven't looked into what changes from the regular build to the doctest one.

**Comment by ekmett** [comment]
I suppose one option is to make the -fsafe option CPP away the doctests that cause problem.

**Comment by glguy** [comment]
Lens infers safe on 7.8 and 7.10 builds. In the 7.6 and probably before it does not. I haven't pinned down exactly what changed in the libraries used between the 7.6 to 7.8 builds that triggers this.

Getting doctests to build with a "Safe" pragma in Control.Lens simply requires us to add "-DTRUSTWORTHY=1" to the GHC invocation line of doctests.hs

**Comment by ekmett** [comment]
I've added some code to propagate the TRUSTWORTHY flag into the test suite and pass it that way.

My experience with lens on 7.8 and 7.10 differs from yours a bit. I'm not getting safe haskell builds all the time there, so we'll possibly want to coordinate to see what is going on.

**Comment by glguy** [comment]
Adding these trustworthy lines when using an unsafe profunctors fixes safe building on 7.6.3 following the uses of (#.) and (.#)

    diff -r lens-4.12.1/src/Control/Lens/Fold.hs lens-4.12.1-safe/src/Control/Lens/Fold.hs
    5a6
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Getter.hs lens-4.12.1-safe/src/Control/Lens/Getter.hs
    4a5
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Internal/Indexed.hs lens-4.12.1-safe/src/Control/Lens/Internal/Indexed.hs
    7a8,9
    >
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Internal/Setter.hs lens-4.12.1-safe/src/Control/Lens/Internal/Setter.hs
    1a2
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Iso.hs lens-4.12.1-safe/src/Control/Lens/Iso.hs
    4a5
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Lens.hs lens-4.12.1-safe/src/Control/Lens/Lens.hs
    7a8
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Level.hs lens-4.12.1-safe/src/Control/Lens/Level.hs
    3a4
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Review.hs lens-4.12.1-safe/src/Control/Lens/Review.hs
    2a3
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Setter.hs lens-4.12.1-safe/src/Control/Lens/Setter.hs
    5a6
    > {-# LANGUAGE Trustworthy #-}
    diff -r lens-4.12.1/src/Control/Lens/Zoom.hs lens-4.12.1-safe/src/Control/Lens/Zoom.hs
    6a7
    > {-# LANGUAGE Trustworthy #-}

**Comment by ekmett** [comment]
So it is mostly a matter of detecting the `profunctors` version and guarding those trustworthy pragmas.

```haskell
#if __GLASGOW_HASKELL__ < 708 || !(MIN_VERSION_profunctors(4,4,0))
{-# LANGUAGE Trustworthy #-}
#endif
```
